src/protobuf-net.Reflection/CSharpCodeGenerator.cs

```diff
@@ -911,6 +913,30 @@ private string GetTypeName(GeneratorContext ctx, string typeName)
             }
         }
 
+        private bool IsNullableType(GeneratorContext ctx, FieldDescriptorProto field)
+        {
+            if (!ctx.IsEnabled("nullablevaluetype"))
+                return false;
+            switch (field.type)
+            {
+                case FieldDescriptorProto.Type.TypeDouble:
+                case FieldDescriptorProto.Type.TypeFloat:
+                case FieldDescriptorProto.Type.TypeBool:
+                case FieldDescriptorProto.Type.TypeSint32:
+                case FieldDescriptorProto.Type.TypeInt32:
+                case FieldDescriptorProto.Type.TypeSfixed32:
+                case FieldDescriptorProto.Type.TypeSint64:
+                case FieldDescriptorProto.Type.TypeInt64:
+                case FieldDescriptorProto.Type.TypeSfixed64:
+                case FieldDescriptorProto.Type.TypeFixed32:
+                case FieldDescriptorProto.Type.TypeUint32:
+                case FieldDescriptorProto.Type.TypeFixed64:
+                case FieldDescriptorProto.Type.TypeUint64:
+                    return true;
+                default:
+                    return false;
```

Comment:
What will happen in case the project is set with Nullable Reference Types?
Maybe in such configuration, all types shall be nullable?

Reply:
Since nullable reference types are supported in C# version 8 and onwards, I think this should be a separate request. @mgravell  thoughts on this matter.